File: 08FlowControl/if_else.cs
using System;
public class Program
{
    public static void Main(string[] args)
    {
        Console.WriteLine("This is C# Language");

        var n = 5;

        if (n < 0) {
            Console.Write("{0} is negative", n);
        } else if (n > 0) {
            Console.Write("{0} is positive", n);
        } else {
            Console.Write("{0} is zero", n);
        }

        //这里还是不如rust，语句分段了。
        int big_n =  n < 10 && n > -10? 10 *n : n/2;
        // 根据条件输出相应的信息
        Console.WriteLine((n < 10 && n > -10) ? ", and is a small number, increase ten-fold"
                                               : ", and is a big number, half the number");
        //   ^ 不要忘记在这里加上一个分号！所有的 `let` 绑定都需要它。

        Console.WriteLine("{0} -> {1}", n, big_n);
    }
}
/* output
5 is positive, and is a small number, increase ten-fold
5 -> 50
*/

File: 08FlowControl/if_else.rs
fn main() {
    let n = 5;
	//简单用法，在C，Python，C#中都有类似的用法。
    if n < 0 {
        print!("{} is negative", n);
    } else if n > 0 {
        print!("{} is positive", n);
    } else {
        print!("{} is zero", n);
    }
	
	//python 可以写成 big_n = 10 * n if n < 10 && n > -10  else  n/2
    let big_n =
        if n < 10 && n > -10 {
            println!(", and is a small number, increase ten-fold");

            // 这个表达式返回一个 `i32` 类型。不能带; 带了就是（）
            10 * n
        } else {
            println!(", and is a big number, half the number");

            // 这个表达式也必须返回一个 `i32` 类型。
            n / 2
            // 试一试 ^ 试着加上一个分号来结束这条表达式。
        };
    //   ^ 不要忘记在这里加上一个分号！所有的 `let` 绑定都需要它。

    println!("{} -> {}", n, big_n);
}
